src/python/grpcio/grpc/_cython/_cygrpc/aio/iomgr/socket.pyx.pxi

```diff
@@ -106,25 +111,35 @@ cdef class _AsyncioSocket:
                 <grpc_error*>0
             )
 
+        self._io_loop.io_mark()
+
     cdef void connect(self,
                       object host,
                       object port,
                       grpc_custom_connect_callback grpc_connect_cb):
         assert not self._reader
         assert not self._task_connect
 
-        self._task_connect = asyncio.ensure_future(
-            asyncio.open_connection(host, port)
-        )
-        self._grpc_connect_cb = grpc_connect_cb
-        self._task_connect.add_done_callback(self._connect_cb)
+        def callback():
+            self._task_connect = asyncio.ensure_future(
+                asyncio.open_connection(host, port),
+                loop=self._loop
+            )
+            self._grpc_connect_cb = grpc_connect_cb
+            self._task_connect.add_done_callback(self._connect_cb)
+
+        self._loop.call_soon_threadsafe(callback)
```

Comment:
What's the performance impact? Alternatively, we can have a wrapper that handles the in-thread scheduling or cross-thread scheduling.

Reply:
Almost negligible, more info here [1]

[1] https://github.com/grpc/grpc/pull/22076/files#r381595041